broke-broker/src/main/java/org/dsngroup/broke/broker/channel/handler/MessageParseHandler.java

```diff
@@ -0,0 +1,63 @@
+package org.dsngroup.broke.broker.channel.handler;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.util.ReferenceCountUtil;
+import org.dsngroup.broke.protocol.Message;
+import org.dsngroup.broke.protocol.MessageBuilder;
+
+import java.util.ArrayDeque;
+
+public class MessageParseHandler extends ChannelInboundHandlerAdapter {
+
+    /**
+     * Read the message from channel, parse to Message instance and send to next handler.
+     * @param ctx {@see ChannelHandlerContext}
+     * @param msg The message of the channel read
+     * */
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        ByteBuf incoming = (ByteBuf)msg;
+        StringBuilder packet = new StringBuilder();
+        ArrayDeque<Message> newMessages = new ArrayDeque();
+
+        try {
+            while(incoming.isReadable()) {
+                packet.append((char)incoming.readByte());
+            }
+
```

Comment:
You may further investigate the message decoder/encoder mechanism in the future.

Reply:
oops, the **netty** message decoder/encoder mechanism.